### DISCUSSION TITLE
Add support for recognizing `lightning` param in BIP 21 QR codes

### DISCUSSION THREAD

**Discussion by sbddesign** [GaloyMoney/galoy-mobile#2058] [Ideas]
## Background
An issue more broadly in the Bitcoin ecosystem is situations where the user must choose whether to generate an on-chain Bitcoin address or a lightning invoice. Bitcoin Beach wallet falls into this category since it supports both on-chain and lightning.

While Bitcoin Beach will scan a QR code of a BIP 21 payment URI, it will only recognize the on-chain address and not a `lightning` parameter containing a BOLT 11 invoice. I would love it if Bitcoin Beach wallet could support scanning a BIP-21 QR with a `lightning` param.

## Steps to Replicate

![image](https://user-images.githubusercontent.com/2414177/158688145-7fe10237-1e40-4608-9337-88fbe14a5f3f.png)

```
bitcoin:bc1qylh3u67j673h6y6alv70m0pl2yz53tzhvxgg7u?amount=0.00001&label=sbddesign%3A%20For%20lunch%20Tuesday&message=For%20lunch%20Tuesday&lightning=lnbc10u1p3pj257pp5yztkwjcz5ftl5laxkav23zmzekaw37zk6kmv80pk4xaev5qhtz7qdpdwd3xger9wd5kwm36yprx7u3qd36kucmgyp282etnv3shjcqzpgxqyz5vqsp5usyc4lk9chsfp53kvcnvq456ganh60d89reykdngsmtj6yw3nhvq9qyyssqjcewm5cjwz4a6rfjx77c490yced6pemk0upkxhy89cmm7sct66k8gneanwykzgdrwrfje69h9u5u0w57rrcsysas7gadwmzxc8c6t0spjazup6
```

- Open Bitcoin Beach wallet
- Scan the above QR code
- Results: it takes me to the payment screen and fills the `To` field with the on-chain address.
- Expected results: it should instead fill the `To` field with the BOLT 11 invoice from the `lightning` param in the QR code (or, it could prompt the user to choose which to use, but I think it would be easier on the user if it defaulted to lightning).

[See this website](https://bitcoinqr.dev/) for more background information.

## Screenshots

![_bitcoin-beach_1](https://user-images.githubusercontent.com/2414177/158688878-3de7771f-9e23-458e-8404-3bedb4c85613.png)

![_bitcoin-beach_2](https://user-images.githubusercontent.com/2414177/158688882-bdfc38a8-111a-42af-bdb3-6065c8898e93.png)

**Comment by nicolasburtey** [comment]
great feedback. we should indeed select the lightning invoice.

a more complex logic would be to suggest a payment on chain if the lightning payment were to fail.

**Comment by openoms** [comment]
Agree with the OP, the default should be the LNpayment, but the best would be the choice between paying on LN and onchain if both are offered (LN still as default).

BTCPay v1.7.0 introduced BIP21 payment codes eg on signet from: https://signet.demo.btcpayserver.org/apps/2SCdQhwFm464BcHxXn44kRVg3iCr/pos
```
bitcoin:TB1QXL9JKN9NV04SMTNJ3KTTY8ZSCXY9YLFMHET6JP?amount=0.00005824&lightning=LNTBS58240N1P3C3X4TPP5ZXNH9CF84G32GX7JUTP56DZ5E4V4R8GTQ46M2W60URYV8MVEP9DQDRD2PSKJEPQW3HJQ5MFVAHX2APQW3JHXAPQ9P8HYER9WGSYJ3P6YPCX7UEDV9C8QHEJ2DPKG5TGWARX6DPKX3PXXJRCTPHRGDRT2FTXWVMFGDEZJCQZPGXQZURSP5ZKV7LQ4RSDCU280KC67EUSUT66RG233NFGZQAQU8TAC8H7Y76NZS9QYYSSQCL7ACMDQF2CZ5EEMD9M8LDCAFXMKEZ33KJ3G5TUKVF2G5EY7LTSH976GZ7DJYXXTUR9DYE5YKT6QV0PRDE5N4LCK8GDFVLD5ERU9WSQQKS3RMP
```
Also the new Zeus v0.7.0 release candidate offers to show BIP21/unifiedQRcode links: https://github.com/ZeusLN/zeus/releases

A good table of wallet compatibility here https://bitcoinqr.dev/ from @agbegin.

Jan 2023 update: Muun Wallet adds Unified QR codes - not by default, can chosse to use in Settings - Advanced settings - Lightning Network - Receiving protocol

**Comment by TheBlueMatt** [comment]
Any update here? At least on the parsing/sending side it would be nice to support the standard most wallets use today!
